### PR TITLE
Add CMake build system, works Win32

### PR DESCRIPTION
**Pull request by robinrowe**
Created CMake build system for pforth. Tested on Win32 with MSVC 2019.

Because I always .gitignore build/ directories, I renamed your ./build/ to ./platform/. I used cmaker to create my cmake files. That automatically writes some doc text files. If you don't want any of those changes, all you need is these 3 files:

CMakeLists.txt
csrc/CMakeLists.txt
csrc/sources.cmake

Robin

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,13 +1,14 @@
-build/unix/*.eo
-build/unix/*.o
-build/unix/pfdicdat.h
-build/unix/pforth
-build/unix/pforth.dic
-build/unix/pforth_standalone
-build/win32/**/.vs
-build/win32/**/Debug
-build/win32/**/Release
+platform/unix/*.eo
```
Comment (philburk) [inline .gitignore:1]: I like changing build to platform.
The use of "build" was a very old decision and wrong in hindsight.

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+if(UNISTD)
+	set(LIBUNISTD_PATH /code/github/libunistd)
+	if(WIN32)
+		include_directories(${LIBUNISTD_PATH}/unistd)
+		link_directories(${LIBUNISTD_PATH}/build/unistd/Release)
+		link_libraries(libunistd)
+	endif(WIN32)
+endif(UNISTD)
+
+option(FLTK "Enable FLTK" false)
```
Comment (philburk) [inline CMakeLists.txt:40]: What is FLTK?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#README.md
+
```
Comment (philburk) [inline README.md:2]: This will override our readme.txt.
We should move readme.txt into README.md in another CL.

File: CMakeLists.txt [hunk 4]
```diff
@@ -2,6 +2,13 @@
 # Created by Robin Rowe 2020-05-01
 # License: License Public Domain
 
+# The process of building pForth involves several steps. 
+# cmake .. 
+# cd fth
+# ../build/Win32/Debug/pforth -i system.fth
```
Comment (philburk) [inline CMakeLists.txt:8]: Can cmake do this?

Comment (robinrowe) [reply CMakeLists.txt:8]: # mkdir -p build/x64
# cd build/x64
# cmake ../.. -A x64
# msbuild Pforth.sln

For 32-bit build, instead of x64 say Win32. 

There is an alternative to msbuild that does about the same thing but isn't as verbose:

# devenv /build Debug Pforth.sln

To make cmake do a post-build step would use add_custom_command() or add_custom_target depending upon the intention.

It is possible to use cmake with the --build switch to call the platform's build too implicitly. I haven't tried it. Is something like this:

# cmake --build . --target ALL_BUILD --config Release -- /nologo /verbosity:minimal /maxcpucount 

By the way, because cmake is normally used to build out of source tree, it will not collide if you leave your legacy make file in your repo for folks who may be using that.

File: CMakeLists.txt (hunk 2 above)
Comment (robinrowe) [reply CMakeLists.txt:40]: FLTK is a portable GUI lib. libunistd is a Windows POSIX library. Since both libs are set to option = false, neither is being used and may be deleted from cmake file as you wish.

File: README.md (hunk 3 above)
Comment (robinrowe) [reply README.md:2]: Take from my README.md anything you find useful. I don't care about authorship.

**Comment by philburk** [comment]
I tried to run this on MacOS. The command "cmake .." above did not work. I tried "cmake ." and got:

    CMake Error at CMakeLists.txt:12 (cmake_minimum_required):
      CMake 3.8 or higher is required.  You are running version 3.6.2

I changed the min to 3.6 and cmake ran OK.
Then the build failed with 

    Undefined symbols for architecture x86_64:
      "_sdQueryTerminal", referenced from:
          _pfCatch in libPforth_lib.a(pf_inner.c.o)

**Comment by philburk** [comment]
On MacOS I had to add the following to csrc/CMakeLists.txt

    if(APPLE)
        set(PLATFORM posix/pf_io_posix.c stdio/pf_fileio_stdio.c)
    endif(APPLE)

File: pforth.cpp [hunk 5]
```diff
@@ -0,0 +1,27 @@
+// pforth.cpp
```
Comment (philburk) [inline pforth.cpp:1]: Is this file necessary?

**Comment by philburk** [comment]
I took your advice and pulled the most critical files into a new PR #92 
I also removed FLTK
and added support for Linux and Mac OS.
